force-app/main/default/classes/DMLOperations.cls

```diff
@@ -145,11 +163,47 @@ public with sharing class DMLOperations {
      * @param accountName The name of the Account.
      * @param oppNames The list of Opportunity names.
      */
+  
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
+        Account acct;
+        
+        List<Account> accounts = [SELECT Id FROM Account WHERE Name = :accountName];
+        if (!accounts.isEmpty()) {
+            acct = accounts[0]; 
+        } else {
+            // If account isn't found, create and insert it
+            acct = new Account(Name = accountName);
+            insert acct;
+        }
+        
+
+        // Find existing opportunities for the account and add to Map: OppName => Opp
+        Map<String, Opportunity> existingOpps = new Map<String, Opportunity>();
```

Comment:
Just like you have the existingOppsList below, you can also name this variable existingOppsMap to tell someone right away what the variable is doing.